addons/misra.py

```diff
@@ -2281,100 +2305,101 @@ def parseDump(self, dumpfile):
-
-        # Under normal operation exit with a non-zero exit code
-        # if there were any violations.
-        if not VERIFY:
-            number_of_violations = len(checker.get_violations())
-            if number_of_violations > 0:
-                exitCode = 1
-
-                if SHOW_SUMMARY:
-                    print("\nMISRA rule violations found: %s\n" % ("\t".join([ "%s: %d" % (viol, len(checker.get_violations(viol))) for viol in checker.get_violation_types()])))
-                    rules_violated = {}
-                    for severity, ids in checker.get_violations():
-                        for misra_id in ids:
-                            rules_violated[misra_id] = rules_violated.get(misra_id, 0) + 1
-                    print("Misra rules violated:")
-                    convert = lambda text: int(text) if text.isdigit() else text
-                    misra_sort = lambda key: [ convert(c) for c in re.split('[\.-]([0-9]*)', key) ]
-                    for misra_id in sorted(rules_violated.keys(), key=misra_sort):
-                        res = re.match(r'misra-c2012-([0-9]+)\\.([0-9]+)', misra_id)
-                        if res is None:
-                            num = 0
-                        else:
-                            num = int(res.group(1)) * 100 + int(res.group(2))
-                        severity = '-'
-                        if num in checker.ruleTexts:
-                            severity = checker.ruleTexts[num].severity
-                        print("\t%15s (%s): %d" % (misra_id, severity, rules_violated[misra_id]))
 
-        if args.show_suppressed_rules:
-            checker.showSuppressedRules()
+def get_args():
+    """Generates list of command-line arguments acceptable by misra.py script."""
+    parser = cppcheckdata.ArgumentParser()
+    parser.add_argument("--rule-texts", type=str, help=RULE_TEXTS_HELP)
+    parser.add_argument("--suppress-rules", type=str, help=SUPPRESS_RULES_HELP)
+    parser.add_argument("--quiet", help="Only print something when there is an error", action="store_true")
+    parser.add_argument("--no-summary", help="Hide summary of violations", action="store_true")
+    parser.add_argument("-verify", help=argparse.SUPPRESS, action="store_true")
+    parser.add_argument("-generate-table", help=argparse.SUPPRESS, action="store_true")
+    parser.add_argument("dumpfile", nargs='*', help="Path of dump file from cppcheck")
+    parser.add_argument("--show-suppressed-rules", help="Print rule suppression list", action="store_true")
+    parser.add_argument("-P", "--file-prefix", type=str, help="Prefix to strip when matching suppression file rules")
+    parser.add_argument("--cli", help="Addon is executed from Cppcheck", action="store_true")
+    return parser.parse_args()
+
+
+def main():
+    args = get_args()
+    settings = MisraSettings(args)
+    checker = MisraChecker(settings)
+
+    if args.generate_table:
+        generateTable()
```

Comment:
I would like a early return here.

Reply:
Done. I suppose we can exit with 'success' return code here?